openedx/core/djangoapps/user_api/accounts/permissions.py

```diff
@@ -13,3 +14,16 @@ class CanDeactivateUser(permissions.BasePermission):
     """
     def has_permission(self, request, view):
         return request.user.has_perm('student.can_deactivate_users')
+
+
+class CanRetireUser(permissions.BasePermission):
+    """
+    Grants access to the various retirement API endpoints if the requesting user is
+    a superuser, the RETIREMENT_SERVICE_USERNAME, or has the explicit permission to
+    retire a User account.
+    """
+    def has_permission(self, request, view):
+        return (
+            request.user.username == settings.RETIREMENT_SERVICE_WORKER_USERNAME or
+            request.user.has_perm('student.can_retire_users')
```

Comment:
Should this permission be defined somewhere?

Reply:
Possibly, but it's just a nice-to-have. I think I'll actually get rid of it for now.